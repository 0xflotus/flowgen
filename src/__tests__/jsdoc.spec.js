// @flow

import { compiler, beautify } from "..";

it("should handle basic jsdoc", () => {
  const ts = `const skip: number
/**
 * @param {string} userToken User authentication token
 * @returns {Promise<void>}
 */
declare function authorize(userToken: string): Promise<void>

/**
 * Plain comment
 */
declare function test(): Promise<void>
`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});

it("should remove jsdoc", () => {
  const ts = `
/**
 * @param {string} userToken User authentication token
 * @returns {Promise<void>}
 */
declare function authorize(userToken: string): Promise<void>
`;
  const result = compiler.compileDefinitionString(ts, { jsdoc: false });
  expect(beautify(result)).toMatchSnapshot();
});
