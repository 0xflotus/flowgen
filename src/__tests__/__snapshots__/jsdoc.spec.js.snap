// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle basic jsdoc 1`] = `
"declare var skip: number;
/**
 * @param {string} userToken User authentication token
 * @returns {Promise<void>}
 */
declare function authorize(userToken: string): Promise<void>;

/**
 * Plain comment
 */
declare function test(): Promise<void>;
"
`;

exports[`should remove jsdoc 1`] = `
"declare function authorize(userToken: string): Promise<void>;
"
`;
